tests/queries/0_stateless/01602_insert_into_table_function_cluster.sql

```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS default.x;
+
+CREATE TABLE default.x AS system.numbers ENGINE = Log;
+
+INSERT INTO FUNCTION cluster('test_shard_localhost', default, x) SELECT * FROM numbers(10);
+-- In fact, in this case(just one shard), sharding key is not required
+INSERT INTO FUNCTION cluster('test_shard_localhost', default, x, rand()) SELECT * FROM numbers(10);
+
+INSERT INTO FUNCTION remote('127.0.0.1', default, x, rand()) SELECT * FROM numbers(10);
+
+INSERT INTO FUNCTION remote('127.0.0.1', default, x, 'default', rand()) SELECT * FROM numbers(10);
```

Comment:
The test should use multiple nodes to check that sharding_key in remote/cluster works

```suggestion
INSERT INTO FUNCTION remote('127.0.0.{1,2}', default, x, rand()) SELECT * FROM numbers(10);

INSERT INTO FUNCTION remote('127.0.0.{1,2}', default, x, 'default', rand()) SELECT * FROM numbers(10);
```